File: leetCode/fukun/101-200/lc_198.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function (nums) {
    if (nums.length === 0) return 0;
    let dp = [nums[0]];
    for (let i = 1; i < nums.length; i++) {
        dp[i] = i > 1 ? Math.max(dp[i - 2] + nums[i], dp[i - 1]) : Math.max(nums[0], nums[1]);
    }
    return dp.pop();
};
console.log(rob([1, 2, 3, 1]));//4

File: leetCode/lwc/lc_198.java
package leetcode;

/**
 * @author lwc
 * @date 2019/11/17 21:56
 */
public class lc_198 {
    public int rob(int[] nums) {
        if(nums.length == 0) return 0;
        int[] v = new int[nums.length];
        if(nums.length > 0) v[0] = nums[0];
        if(nums.length > 1) v[1] = Math.max(v[0], nums[1]);
        for(int i = 2; i < nums.length; i++){
            v[i] = Math.max(v[i-1], v[i-2] + nums[i]);
        }
        return v[nums.length - 1];
    }
}
